src/Model/Entity/ReadTrait.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Shim\Model\Entity;
+
+use ArrayAccess;
+use Cake\Datasource\EntityInterface;
+
+/**
+ * Trait to read entity properties in a way that it doesnt throw exception if parts of the path are null.
+ *
+ * - read('tags.0.name') will return the property if it exists or returns null otherwise
+ *
+ * Note: This is primarily for convenience usage in view templates, as the return type cannot be annotated here.
+ *
+ * @mixin \Cake\ORM\Entity
+ */
+trait ReadTrait {
+
+	/**
+	 * Performant iteration over the path to nullable read the property path.
+	 *
+	 * Note: Hash::get($this->toArray(), $path, $default); would be simpler, but slower.
+	 *
+	 * @param string $path
+	 * @param mixed $default The return value when the path does not exist
+	 * @return mixed|null The value fetched from the entity, or null.
+	 */
+	public function read($path, $default = null) {
+		if (isset($this->$path)) {
+			return $this->$path;
+		}
+
+		if (strpos($path, '.') === false) {
+			return $default;
+		}
+
+		$parts = explode('.', $path);
+
+		$data = null;
+		foreach ($parts as $key) {
+			if ($data === null && !isset($this->$key)) {
+				return $default;
+			}
+			if ($data === null) {
+				$data = $this->$key;
+				continue;
+			}
+
+			if ($data instanceof EntityInterface) {
+				$data = $data->toArray();
```

Comment:
why converting to array ?

Reply:
Do u know an Improvement here?

Reply:
i did not test it yet and I think this (if ) is redundant
if its `EntityInterface` just `$data = $data->key` and `continue;`  
i think this could  works without future side-effect

I can test and PR if it was successful :)

Reply:
also for php 7 > , you can do like 
```php
		if (version_compare(PHP_VERSION, '7.0.0') >= 0) {
			try{
				$result = $entity->get('c1')->get('c2')->get('c3');
			} catch (\Throwable $e) {
				$result = null;
			}
			return $result;
		}
```